ldms/src/sampler/zfs_leafvdevs/zfs_leafvdevs.c

```diff
@@ -302,15 +304,19 @@ struct ldmsd_plugin *get_plugin(ldmsd_msg_log_f pf)
  *  init_errors
  */
 static int get_leafvdev_stats(nvlist_t *nvroot, const char *pool_name,
-    const char *parent_name)
+    const char *parent_name, zpool_handle_t *zhp)
 {
         uint_t c;
         vdev_stat_t *vs;
         char *vdev_name = NULL;
+	char *vdev_path = NULL;
+	char *vdev_guid = NULL;
 	ldms_mval_t record_instance;
         int rc=0; /*return code*/
 
-	vdev_name = get_vdev_name(nvroot, parent_name);
+	vdev_name = zpool_vdev_name(g_zfs, zhp, nvroot, 0);
+	vdev_path = zpool_vdev_name(g_zfs, zhp, nvroot, VDEV_NAME_PATH | VDEV_NAME_FOLLOW_LINKS);
+	vdev_guid = zpool_vdev_name(g_zfs, zhp, nvroot, VDEV_NAME_GUID);
```

Comment:
This memory will be leaked.  Maybe that's fine if the sampler is intended to be short lived.  I'm not sure what the convention is for other samplers.